extension/src/popup/Router.tsx

```diff
@@ -227,10 +227,13 @@ const HomeRoute = () => {
 const RouteListener = () => {
   const dispatch = useDispatch();
   const location = useLocation();
+  const settingsState = useSelector(settingsStateSelector);
 
   useEffect(() => {
-    dispatch(navigate(location));
-  }, [dispatch, location]);
+    if (settingsState === SettingsState.SUCCESS) {
+      dispatch(navigate(location));
```

Comment:
This action is used to emit `loaded screen:` events. None of them were firing on initial page load because the settings weren't loaded yet. When a user would nav to a different view, this would fire correctly bc the settings would have loaded by then. But you would lose the initial page load, which is important for views like `/sign-transaction` that pop up from freighter-api

Reply:
This `RouteListener` used to instantiate after all account and settings were loaded